gnocchi/rest/aggregates/measures_grouper.py

```diff
@@ -0,0 +1,326 @@
+        grouped_values = \
+            itertools.groupby(
+                to_group,
+                lambda x: tuple((attr, x[attr]) for attr in self.groups))
+
+        grouped = []
+        for key, values in grouped_values:
+            resources = []
+            for value in values:
+                self.truncate_resource_time_window(value)
+                resources.append(value)
+            grouped.append(MeasureGroup(key, resources))
+
+        return grouped
+
+    def truncate_resource_time_window(self, value):
+        if value['revision_start']:
+            value['revision_start'] = numpy.datetime64(value['revision_start'],
+                                                       dtype='datetime64[ns]')
+            if value['revision_start'] < self.start:
+                LOG.debug("[Truncating %s revision_start to %s]",
+                          value['id'], self.start)
+                value['revision_start'] = self.start
+        else:
+            value['revision_start'] = self.start
+
+        if value['revision_end']:
+            value['revision_end'] = numpy.datetime64(value['revision_end'],
+                                                     dtype='datetime64[ns]')
+            if value['revision_end'] > self.end:
+                LOG.debug("[Truncating %s revision_end to %s]",
+                          value['id'], self.end)
+                value['revision_end'] = self.end
+        else:
+            value['revision_end'] = self.end
+
+    def get_measures(self, groups):
+        for group in groups:
+            first_start = self.start
+            date_map = self.get_date_map(group)
+            for resource in group.resources:
+                start = numpy.datetime64(resource['revision_start'],
+                                         dtype='datetime64[ns]') \
+                    if resource['revision_start'] else None
+                stop = numpy.datetime64(resource['revision_end'],
+                                        dtype='datetime64[ns]') \
+                    if resource['revision_end'] else None
+
+                LOG.debug("[ Collecting measures from %s to %s ]",
+                          start, stop)
+
+                try:
+                    if not first_start:
+                        first_start = start
+                        start = None
+                    measure = _get_measures_by_name(
+                        [resource], self.references, self.body["operations"],
+                        start, stop, self.granularity, self.needed_overlap,
+                        self.fill, details=self.details)
+                    group.add_measures(measure, date_map, start, stop,
+                                       self.details)
+                except indexer.NoSuchMetric:
+                    continue
+
+        self.truncate_measures()
+        self.sum_sequential_group_metrics(groups)
+        return groups
+
+    def get_date_map(self, group):
```

Comment:
This "get_data_map" function will return a map (dict) within all group`s measures grouped by timestamp (date). 

If the group has no measures yet, then it will set the group entry and return the dict reference to be filled with the measures in  the "get_measures" function. 

If the group has no ID them it will return all measures grouped by group and grouped by date